cmd/d2/main.go

```diff
@@ -224,6 +242,13 @@ func compile(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, theme
 	if err != nil {
 		return nil, err
 	}
+
+	// Missing/broken images are fine during watch mode, as the user is likely building up a diagram.
+	// Otherwise, the assumption is that this diagram is building for production, and broken images are not okay.
+	if !isWatching && ms.Log.Nerrors() > 0 {
+		os.Exit(1)
```

Comment:
Never do `os.Exit` outside of `main`, never! Here be dragons 🐉 😈 👹 
Seriously, zero exceptions, absolutely zero.

Instead return `xmain.ExitErrorf(1, "errors logged while rendering, partial output written to %v", outputPath)`